smart_open/s3.py

```diff
@@ -412,8 +457,10 @@ def read(self, size=-1):
             return b''
         elif size < 0:
             from_buf = self._read_from_buffer()
+            # call read() before setting _current_pos to make sure _content_length is set
+            out = from_buf + self._raw_reader.read()
```

Comment:
Why bother defining `from_buf` above, when you can do this:

```suggestion
            out = self._read_from_buffer() + self._raw_reader.read()
```